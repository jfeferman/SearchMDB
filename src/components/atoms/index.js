export { default as Cta } from './Cta';
export { default as Heading } from './Heading';
export { default as TextInput } from './TextInput';
