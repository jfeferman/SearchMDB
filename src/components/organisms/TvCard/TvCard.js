import React from 'react';
import PropTypes from 'prop-types';
import { Heading } from '../../atoms';

const TvCard = (props) => {
  const {
    name,
    first_air_date,
    overview,
  } = props.item;

  return (
    <div>
      <Heading level="3">{name}</Heading>
      <div>
        First aired:
        {' '}
        {first_air_date}
      </div>
      <div>{overview}</div>
    </div>
  );
};

TvCard.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string,
    first_air_date: PropTypes.string,
    overview: PropTypes.string,
  }),
};

export default TvCard;
