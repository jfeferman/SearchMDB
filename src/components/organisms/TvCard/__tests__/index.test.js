import React from 'react';
import 'jest-styled-components';
import { create } from 'react-test-renderer';
import TvCard from '..';
import { Heading } from '../../../atoms';

describe('components/molecules/Search', () => {
  let renderer;
  let instance;
  const item = {
    title: 'A title',
    first_air_date: '01/01/2019',
    overview: 'An overview',
  };

  beforeEach(() => {
    renderer = create(<TvCard item={item} />);
    instance = renderer.root;
  });

  it('matches the snapshot of the component', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('renders a heading', () => {
    expect(instance.findAll(el => el.type === Heading)).toHaveLength(1);
  });
});
