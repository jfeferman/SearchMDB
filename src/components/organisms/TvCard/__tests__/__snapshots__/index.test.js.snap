// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/Search matches the snapshot of the component 1`] = `
.c0 {
  font-weight: 400;
  font-style: normal;
  color: #000000;
  font-size: 1.25rem;
  line-height: 1.875rem;
  -webkit-letter-spacing: 0.034375rem;
  -moz-letter-spacing: 0.034375rem;
  -ms-letter-spacing: 0.034375rem;
  letter-spacing: 0.034375rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.25rem;
    line-height: 1.875rem;
    -webkit-letter-spacing: 0.034375rem;
    -moz-letter-spacing: 0.034375rem;
    -ms-letter-spacing: 0.034375rem;
    letter-spacing: 0.034375rem;
  }
}

<div>
  <h3
    className="c0"
  />
  <div>
    First aired:
     
    01/01/2019
  </div>
  <div>
    An overview
  </div>
</div>
`;
